scripts/generator/db.py

```diff
@@ -109,7 +109,9 @@ def update_or_create_activity(session, run_activity):
                 location_country=location_country,
                 average_heartrate=run_activity.average_heartrate,
                 average_speed=float(run_activity.average_speed),
-                summary_polyline=run_activity.map.summary_polyline,
+                summary_polyline=(
+                    run_activity.map and run_activity.map.summary_polyline or ""
```

Comment:
what is the `and` here?

Reply:
if has `map` member which is a valid object, then  get return`summary_polyline `, otherwise get empty string.

Reply:
IMO
the logic is for the `and` is that
if run_activity.map.summary_polyline is not None or not empty string 
summary_polyline = run_activity.map.summary_polyline
and ignore  run_activity.map am I right?

Reply:
make sence.

Reply:
summary_polyline = run_activity.map.summary_polyline if run_activity.map else ""
maybe more easy to understand?

Reply:
if `run_activity.map` is always not None then can remove run_activity.map validate.

Reply:
@shuxiao9058 
This code is good for me.
I wonder if there is an easier way to understand.

if not will merge it now.

Reply:
merging it now.